### PR TITLE
feat: gates

### PR DESCRIPTION
**Pull request by tracernz**

### CODE REVIEW

File: src/server/index.ts [hunk 1]
```diff
@@ -426,6 +426,17 @@ export function msfsNavdataRouter(provider: NavigraphProvider, development: bool
...
 
+    router.get('/airport/:ident/gates', (req, res) => {
+        try {
+            const ident = parseAirportIdent(req.params.ident);
+            provider.getGates(ident).then((gates) => {
+                res.json(gates);
+            }).catch((error) => errorResponse(error, res));
+        } catch (error) {
+            errorResponse(error, res);
+        }
```
Comment (professoralex13) [inline src/server/index.ts:437]: do we need the ```.catch((error) => errorResponse(error, res));``` when its all wrapped in a try {} catch {} anyway?

actually does that only work when using async? ive forgotten

File: src/shared/types/Gate.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    /** 2-digit ICAO region the airport belongs to */
+    icaoCode: string,
+
+    /** identifier name for the gate, up to 5 alpha-numeric chars */
+    ident: string,
+
+    /** location of the parking position at the gate */
+    location: Coordinates
+}
```
Comment (professoralex13) [inline src/shared/types/Gate.ts]: should this extend DatabaseItem?

File: src/server/index.ts (hunk 1 above)
Comment (tracernz) [reply src/server/index.ts:437]: Actually... if we awaited the getGates promise then no we wouldn't need it. The function we're in isn't async though, so as you alluded to we can't...

File: src/shared/types/Gate.ts (hunk 2 above)
Comment (tracernz) [reply src/shared/types/Gate.ts]: I thought about that, and still am... probably yes.

**Review by professoralex13 — APPROVED** [review]
lgtm

**Pull request merged**